includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php

```diff
@@ -432,49 +533,87 @@ public function processAdopted($profiles, $orphans) {
-    // Set the flag to 1 in the capx_profiles table.
-
-    // BEAN is returning its delta when using this.
-    // $id = $entity->getIdentifier();
-
-    $entityType = $entity->type();
-    $entityRaw = $entity->raw();
-    list($id, $vid, $bundle) = entity_extract_ids($entityType, $entityRaw);
-
+  public function processAdoptedMultiple($profiles) {
     $importer = $this->getImporter();
-    $importerName = $importer->getMachineName();
+    $importerMachineName = $importer->getMachineName();
     $entityType = $importer->getEntityType();
     $bundleType = $importer->getBundleType();
+    $options = $importer->getOptions();
+    $orphanAction = $options['orphan_action'];
 
-    $record = array(
-      'entity_type' => $entityType,
-      'bundle_type' => $bundleType,
-      'entity_id' => $id,
-      'importer' => $importerName,
-      'orphaned' => 1,
-    );
+    if ($orphanAction !== "unpublish") {
+      // Nothing actionable to perform.
+      return;
+    }
 
-    $keys = array(
-      'entity_type',
-      'entity_id',
-      'importer',
-      'bundle_type',
-    );
+    $results = $this->getResults();
+    $mapper = $this->getImporter()->getMapper();
+    $guuidquery = $mapper->getGUUIDQuery();
+    $parts = explode(".", $guuidquery);
+    $subquery = "$.." . array_pop($parts);
+    $remoteGUUIDs = $mapper->getRemoteDataByJsonPath($results, $subquery);
+
+    $or = db_or()->condition('profile_id', $profiles);
+    $localResults = db_select("capx_profiles", "cxp")
+      ->fields('cxp', array('entity_id', 'guuid'))
+      ->condition($or)
+      ->condition("importer", $importerMachineName)
+      ->condition("orphaned", 1)
+      ->execute()
+      ->fetchAllAssoc('guuid');
+
+    $localGUUIDs = array_keys($localResults);
+    $diff = array_intersect($localGUUIDs, $remoteGUUIDs);
+
+    // Nothing to adopt. Yay.
+    if (empty($diff)) {
+      return array();
+    }
 
-    $yes = drupal_write_record('capx_profiles', $record, $keys);
+    foreach ($diff as $guuid) {
+      $entityID = $localResults[$guuid]->entity_id;
+      $ids = array($entityID);
+      $entity = array_pop(entity_load($entityType, $ids));
+      $profile = entity_metadata_wrapper($entityType, $entity);
+      $profile->status->set(1);
+      $profile->save();
+
+      // Profile is not an orphan. Lets enable it again.
+      $record = array(
+        'entity_type' => $entityType,
+        'bundle_type' => $bundleType,
+        'entity_id' => $entityID,
+        'importer' => $importerMachineName,
+        'orphaned' => 0,
+        'guuid' => $guuid,
+      );
 
-    if ($yes) {
-      watchdog('EntityImporterOrphans', "%title was orphaned from the importer %importername.", array("%title" => $entity->label(), "%importername" => $importerName), WATCHDOG_NOTICE, '');
+      $keys = array(
+        'entity_type',
+        'entity_id',
+        'importer',
+        'bundle_type',
+        'guuid',
+      );
+
+      drupal_write_record('capx_profiles', $record, $keys);
```

Comment:
I'm not seeing a discrepancy in the data browser:
![Shows the value](https://cloud.githubusercontent.com/assets/550602/16162421/2ca17914-3488-11e6-9c92-29f577451579.png)